### ISSUE TITLE
bug(linter) false positive in no-unused-vars

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#5274]
```ts
export const render = markdown => {
  const rendered = marked(markdown, {
    renderer: new (class CustomRenderer extends Renderer {
      heading({ tokens }) {
        return `
          <Fragment>
            <Section><b>${tokens[0].text}</b></Section>
            <Divider />
          </Fragment>`;
      }

      list(token) {
        return `<Section>${super.list(token)}</Section>`;
      }

      hr() {
        return `<Divider />`;
      }
    })(),
  });
  return jsxslack([`<Blocks>${rendered}</Blocks>`]);
};
```

```
  × eslint(no-unused-vars): Class 'CustomRenderer' is declared but never used.
   ╭─[tools/changelog/markdown.js:6:26]
 5 │   const rendered = marked(markdown, {
 6 │     renderer: new (class CustomRenderer extends Renderer {
   ·                          ───────┬──────
   ·                                 ╰── 'CustomRenderer' is declared here
 7 │       heading({ tokens }) {
   ╰────
  help: Consider removing this declaration.

Finished in 113ms on 2036 files with 99 rules using 12 threads.
Found 0 warnings and 1 error.
```

**Comment by DonIsaac** [comment]
Man I just love JavaScript sometimes

**Comment by Boshen** [comment]
I'll make a new oxlint release once this is fixed for Affine https://github.com/oxc-project/oxlint-ecosystem-ci/actions/runs/10591385888/job/29348827098
